drivers/adc/adc_ad559x.c

```diff
@@ -272,24 +290,31 @@ static int adc_ad559x_init(const struct device *dev)
 	return 0;
 }
 
-static const struct adc_driver_api adc_ad559x_api = {
-	.channel_setup = adc_ad559x_channel_setup,
-	.read = adc_ad559x_read,
 #ifdef CONFIG_ADC_ASYNC
-	.read_async = adc_ad559x_read_async,
+#define ADC_AD559X_ASYNC() .read_async = adc_ad559x_read_async,
+#else
+#define ADC_AD559X_ASYNC()
 #endif
-	.ref_internal = AD559X_ADC_VREF_MV,
```

Comment:
I took a look at this part again and I'm worried why you didn't do it this way:

<pre>
static const struct adc_driver_api adc_ad559x_api = {
	.channel_setup = adc_ad559x_channel_setup,
	.read = adc_ad559x_read,
#ifdef CONFIG_ADC_ASYNC
	.read_async = adc_ad559x_read_async,
#endif
	.ref_internal = AD559X_ADC_VREF_MV * (1 + DT_INST_PROP(inst, double_input_range)),
};
</pre>

Then you could remove entire `ADC_AD559X_DRIVER_API` and etc...

Reply:
Because different instances can have different `.ref_internal`.

Reply:
You are right, I have forgotten about that.